File: Solutions/_1768.cpp
#include <iostream>
#include <string>

std::string solve(std::string word1, std::string word2){
    std::string result = "";
    int longest = word1.length() > word2.length() ? word1.length() : word2.length();
    for (int i = 0; i < longest; i++){
        if (i < word1.length()){
            result += word1[i];
        }
        if (i < word2.length()){
            result += word2[i];
        }
    }

    return result; 
}

int main(){
    std::string word1 = "abcd";
    std::string word2 = "pq";

    std::string result = solve(word1, word2);
    std::cout << result << "\n";
}

File: Solutions/_1768.go
func mergeAlternately(word1 string, word2 string) string {
	i := 0
	j := 0
	result := ""

	for i < len(word1) && j < len(word2) {
		result = result + string(word1[i])
		result = result + string(word2[j])
		i++
		j++
	}

	result += word1[i:]
	result += word2[j:]

	return result
}
